extensions/amp-ad/0.1/amp-ad-custom.js

```diff
@@ -84,7 +84,9 @@ export class AmpAdCustom extends AMP.BaseElement {
     // If this promise has no URL yet, create one for it.
     if (!(fullUrl in ampCustomadXhrPromises)) {
       // Here is a promise that will return the data for this URL
-      ampCustomadXhrPromises[fullUrl] = xhrFor(this.win).fetchJson(fullUrl);
+      ampCustomadXhrPromises[fullUrl] = xhrFor(this.win).fetchJson(fullUrl, {
```

Comment:
Let me know when you've put in a separate issue. In the mean time, I've added the following header to the source code of my ad server:
`AMP-Access-Control-Allow-Source-Origin: *`

What I would really like to know is this: do you have any documents explaining the kind of attack that this feature is designed to protect against? Because in this case, where the response from the server is never interpreted as a script, I can't immediately think of one...